src/Form/Formbot.js

```diff
@@ -110,15 +114,27 @@ export default class Formbot extends React.Component {
 
   validateField(field) {
     return new Promise(resolve => {
-      const validation = this.props.validations[field];
+      const fieldState = this.state.fields[field] || {};
+      if (fieldState.validated) {
+        resolve();
+        return;
+      }
 
-      if (!validation) return;
+      const fromSchema = !!this.props.validationSchema;
+      const validation = (fromSchema ? this.props.validationSchema : this.props.validations)[field];
+
+      if (!validation) {
+        resolve();
+        return;
+      }
 
       const fieldValue = this.state.values[field];
       let errorMsg;
 
       try {
-        if (typeof validation === 'function') {
+        if (fromSchema) {
```

Comment:
I see we're reusing `fromSchema` here so that we don't have to perform the `!!this.props.validationSchema` check twice. If this is the only instance, can we perform the check inline?